dmap.js

```diff
@@ -1,5 +1,10 @@
 module.exports = lib = {}
 
+const multiformats = require('multiformats')
+const prefLenIndex = 29
```

Comment:
I think this is the last thing to fix
Flags and metadata should be on opposite sides of the `flags` field (implies we need to change one of these names)
`[value metadata -->  ...  <-- flags]`

Reply:
We should also strongly consider inverting this